tests/Aspire.EndToEnd.Tests/IntegrationServicesFixture.cs

```diff
@@ -151,15 +287,30 @@ public async Task DisposeAsync()
         }
     }
 
-    private static string GetRepoRoot()
+    public void EnsureAppHostRunning()
     {
-        var directory = AppContext.BaseDirectory;
+        if (_appHostProcess is null || _appHostProcess.HasExited || _appExited.Task.IsCompleted)
+        {
+            throw new InvalidOperationException("The app host process is not running.");
+        }
+    }
 
-        while (directory != null && !Directory.Exists(Path.Combine(directory, ".git")))
+    private static IList<string> GetResourcesToSkip()
+    {
+        List<string> resourcesToSkip = new();
```

Comment:
```suggestion
        HashSet<string> resourcesToSkip = new();
```

? We don't really want dupes in this collection.